### PR TITLE
Reproduce pay route crash

### PR DESCRIPTION
**Pull request by rustyrussell**
Reproduces the crash in #3607 which @niftynei fixed in #3630

### CODE REVIEW

**Comment by rustyrussell** [comment]
Rebase on master, which contains the fix this tests.

File: tests/test_pay.py [hunk 1]
```diff
@@ -2880,3 +2880,19 @@ def test_reject_invalid_payload(node_factory):
...
+
+def test_excluded_adjacent_routehint(node_factory, bitcoind):
+    """Test case where we try have a routehint which leads to an adjacent
+    node, but the result exceeds our maxfee; we crashed trying to find
+    what part of the path was most expensive in that case
+
+    """
+    l1, l2, l3 = node_factory.line_graph(3)
+
+    # We're be forced to use routehint, since we don't know about l3.
```
Comment (niftynei) [inline tests/test_pay.py:2893]:
```suggestion
    # We'll be forced to use routehint, since we don't know about l3.
```

**Comment by niftynei** [comment]
seems like you've got some whitespace issues in the python and a timing issue. lgtm otherwise.

```
>           raise RpcError(method, payload, resp['error'])
E           pyln.client.lightning.RpcError: RPC call failed: method: invoice, payload: {'msatoshi': 1000, 'label': 'lbl', 'description': 'desc', 'exposeprivatechannels': '103x2x1'}, error: {'code': 902, 'message': 'None of those hints were suitable local channels'}
```

**Comment by rustyrussell** [comment]
Trivial rebase

**Comment by rustyrussell** [comment]
... and added Changelog into commit msg for deadend heuristic and expostprivatechannels.

**Comment by niftynei** [comment]
ACK https://github.com/ElementsProject/lightning/pull/3633/commits/79c5e0ae5d89563b2406e8dd5009b363bb2957e9

**Pull request merged**